test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
-      var delay = 500;
-
-      sessionPool.reads_ = [{}];
-      sessionPool.destroySession_ = function() {
-        return new Promise(function(resolve) {
-          setTimeout(resolve, delay);
+    describe('release', function() {
+      it('verify that a read and write session are borrowed and then release them', function() {
+        sessionPool.createTransaction_ = () => Promise.resolve();
+        sessionPool.session_ = function() {
+          return new Object({
+            create: () => Promise.resolve(),
+            delete: () => Promise.resolve(),
+          });
+        };
+        sessionPool.open();
+        clearTimeout(sessionPool.pingTimeoutHandle);
+        return Promise.all([
+          sessionPool.getReadSession(),
+          sessionPool.getWriteSession(),
+        ]).then(function(sessions) {
+          const readSession = sessions[0];
+          const writeSession = sessions[1];
+          assert.strictEqual(sessionPool.getStats().writePool.borrowed, 1);
+          assert.strictEqual(sessionPool.getStats().readPool.borrowed, 1);
+          return Promise.all([
+            sessionPool.release(readSession),
+            sessionPool.release(writeSession),
+          ]);
         });
-      };
-
-      var end = timeSpan();
+      });
 
-      return sessionPool.close().then(function() {
-        assert(isAround(delay, end()));
+      it('should throw an error while release a write session', function() {
+        let isSecondCall = false;
+        sessionPool.createTransaction_ = () => {
+          if (isSecondCall) {
+            return Promise.reject({});
+          }
+          isSecondCall = true;
+          return Promise.resolve();
+        };
+        sessionPool.session_ = function() {
+          return new Object({
+            create: () => Promise.resolve(),
+            delete: () => Promise.resolve(),
+          });
+        };
+        sessionPool.open();
+        clearTimeout(sessionPool.pingTimeoutHandle);
+        return sessionPool.getWriteSession().then(session =>
+          sessionPool.release(session).then(() => {
+            assert.strictEqual(sessionPool.getStats().writePool.available, 0);
+          })
+        );
       });
     });
-  });
 
-  describe('fill', function() {
-    var created;
+    describe('request', function() {
+      let CONFIG;
+      const SESSION = {
+        formattedName_: 'formatted-name',
+      };
 
-    beforeEach(function() {
-      sessionPool.options.min = 8;
+      beforeEach(function() {
+        CONFIG = {
+          reqOpts: {},
+        };
 
-      created = {
-        readonly: 0,
-        readwrite: 0,
-      };
+        sessionPool.getReadSession = function() {
+          return Promise.resolve(SESSION);
+        };
 
-      sessionPool.createSessionInBackground_ = function(type) {
-        created[type] += 1;
-      };
-    });
+        sessionPool.release = common.util.noop;
+      });
+
+      it('should get a session', function(done) {
```

Comment:
What is this testing? We don't assert that the session is returned or that the options are passed anywhere.

Reply:
This is not a new test case. I just copied it from existing test cases for session pool